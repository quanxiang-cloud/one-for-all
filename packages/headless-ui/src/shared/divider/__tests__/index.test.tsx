import React from 'react';
import { render } from '@testing-library/react';

import Divider from '../index';

test('Divider component test', () => {
  const node: DividerProps = {
    className: 'test-divider-props',
    size: '10px',
    thickness: '30px',
    style: {
      backgroundColor: 'blue',
    },
  };

  const { container } = render(<Divider {...node} />);
  expect(container).toMatchSnapshot();
});

test('Divider component test', () => {
  const node: DividerProps = {
    className: 'test-divider-props',
    size: '20px',
    direction: 'vertical',
    thickness: '30px',
    style: {
      backgroundColor: 'red',
    },
  };

  const { container } = render(<Divider {...node} />);
  expect(container).toMatchSnapshot();
});

test('Divider component test', () => {
  const ref = React.createRef<HTMLDivElement>();
  render(<Divider ref={ref} />);
  expect(ref.current && ref.current instanceof HTMLDivElement).toBeTruthy();
});
