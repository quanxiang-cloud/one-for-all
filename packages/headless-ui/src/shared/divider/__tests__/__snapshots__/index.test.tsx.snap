// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component test 1`] = `
<div>
  <div
    class="ofa-divider test-divider-props"
    style="height: 30px; width: 10px; background-color: blue;"
  />
</div>
`;

exports[`Divider component test 2`] = `
<div>
  <div
    class="ofa-divider test-divider-props"
    style="width: 30px; height: 20px; background-color: red;"
  />
</div>
`;
